### PR TITLE
DOCS-440: Updated ASOF JOIN description

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #7282

### CODE REVIEW

**Comment by BayoNet** [comment]
@4ertus2 Look at the changes please.

File: docs/en/query_language/select.md [hunk 1]
```diff
@@ -564,29 +564,37 @@ ClickHouse doesn't directly support syntax with commas, so we don't recommend us
...
-    ON equi_cond AND closest_match_cond
-    ```
+Conditions supported for the closest match:
 
-    You can use any number of equality conditions and exactly one closest match condition. For example, `SELECT count() FROM A ASOF LEFT JOIN B ON A.a == B.b AND B.t <= A.t`. Only `table_2.some_col <= table_1.some_col` and `table_1.some_col >= table2.some_col` condition types are available. You can't apply other conditions like `>` or `!=`.
+- `table_1.some_col > table_2.some_col`
+- `table_1.some_col < table_2.some_col`
+- `table_1.some_col >= table_2.some_col`
+- `table_1.some_col <= table_2.some_col`
+- `table_2.some_col <= table_1.some_col`
```
Comment (qoega) [inline docs/en/query_language/select.md]: - In previous version < and > are strictly forbidden
- I don't think we should explicitly write all possible conditions. Only one version with table2 <= table1 makes it more confusing

File: docs/en/query_language/select.md [hunk 2]
```diff
@@ -602,7 +610,7 @@ event_1_2 |  13:00  |  42         event_2_3 |  13:00  |   42
               ...                               ...
 ```
 
-`ASOF JOIN` can take the timestamp of a user event from `table_1` and find an event in `table_2` where the timestamp is closest (equal to or less) to the timestamp of the event from `table_1`. Here, the `user_id` column can be used for joining on equality and the `ev_time` column can be used for joining on the closest match. In our example, `event_1_1` can be joined with `event_2_1` and `event_1_2` can be joined with `event_2_3`, but `event_2_2` can't be joined.
+`ASOF JOIN` can take the timestamp of a user event from `table_1` and find an event in `table_2` where the timestamp is closest to the timestamp of the event from `table_1` corresponding to the closest match condition. Here, the `user_id` column can be used for joining on equality and the `ev_time` column can be used for joining on the closest match. In our example, `event_1_1` can be joined with `event_2_1` and `event_1_2` can be joined with `event_2_3`, but `event_2_2` can't be joined.
```
Comment (qoega) [inline docs/en/query_language/select.md]:
```suggestion
`ASOF JOIN` can take the timestamp of a user event from `table_1` and find an event in `table_2` where the timestamp is closest to the timestamp of the event from `table_1` corresponding to the closest match condition. Equal timestamp values are the closest if available. Here, the `user_id` column can be used for joining on equality and the `ev_time` column can be used for joining on the closest match. In our example, `event_1_1` can be joined with `event_2_1` and `event_1_2` can be joined with `event_2_3`, but `event_2_2` can't be joined.
```
Or something more clear. We should not drop information about possible equal values.

**Review by qoega — CHANGES_REQUESTED** [review]
There is error in ASOF logic - it supports only `>=` and `<=`

File: docs/en/query_language/select.md [hunk 3]
```diff
@@ -564,29 +564,37 @@ ClickHouse doesn't directly support syntax with commas, so we don't recommend us
...
 
-- `ASOF JOIN ... ON`
+```sql
+SELECT expressions_list
+FROM table_1
+ASOF LEFT JOIN table_2
+ON equi_cond AND closest_match_cond
+```
+
+You can use any number of equality conditions and exactly one closest match condition. For example, `SELECT count() FROM A ASOF LEFT JOIN B ON A.a == B.b AND B.t <= A.t`.
```
Comment (4ertus2) [inline docs/en/query_language/select.md]: mix of table_1/table_2 and A/B

File: docs/en/query_language/select.md (hunk 1 above)
Comment (4ertus2) [reply docs/en/query_language/select.md]: I think we could say here: You are able to use one of these conditions `>`, `>=`, `<`, `<=`

File: docs/en/query_language/select.md [hunk 4]
```diff
@@ -564,29 +564,37 @@ ClickHouse doesn't directly support syntax with commas, so we don't recommend us
...
+
+```sql
+SELECT expressions_list
+FROM table_1
+ASOF JOIN table_2
+USING (equi_column1, ... equi_columnN, asof_column)
+```
 
-    `ASOF JOIN` uses `equi_columnX` for joining on equality and `asof_column` for joining on the closest match with the `table_1.asof_column >= table2.asof_column` condition. The `asof_column` column must be the last in the `USING` clause.
+`ASOF JOIN` uses `equi_columnX` for joining on equality and `asof_column` for joining on the closest match with the `table_1.asof_column >= table_2.asof_column` condition. The `asof_column` column must be the last in the `USING` clause.
```
Comment (4ertus2) [inline docs/en/query_language/select.md]: > The `asof_column` column must be the last in the `USING` clause.

I think must is not the best word here. We interpret last column in use as asof column. Others - as ususal equality columns.

**Pull request merged**